app/src/main/java/com/islaharper/dawnofandroid/presentation/common/EmptyScreen.kt

```diff
@@ -0,0 +1,145 @@
+package com.islaharper.dawnofandroid.presentation.common
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.Icon
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.pullrefresh.PullRefreshIndicator
+import androidx.compose.material.pullrefresh.pullRefresh
+import androidx.compose.material.pullrefresh.rememberPullRefreshState
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.paging.LoadState
+import androidx.paging.compose.LazyPagingItems
+import com.islaharper.dawnofandroid.R
+import com.islaharper.dawnofandroid.domain.model.Flavour
+import com.islaharper.dawnofandroid.ui.theme.DawnOfAndroidTheme
+import com.islaharper.dawnofandroid.ui.theme.PADDING_MEDIUM
+import java.net.ConnectException
+import java.net.SocketTimeoutException
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun EmptyScreen(
+    modifier: Modifier = Modifier,
+    image: Int = R.drawable.search_document,
+    message: String = stringResource(R.string.find_a_flavour),
+    errorMessage: LoadState.Error? = null,
+    flavours: LazyPagingItems<Flavour>? = null,
+) {
+    var description by remember {
```

Comment:
the `remember` makes sense for `isRefreshing` but not so much for the other variables `description` and `icon`. It looks like they are being used to try and get the Composable to switch between an error state and some other state. We need to concentrate on making the Composable simple, where the ViewModel has already done some calculation for it, and the Composable will forward to other Composables that are also, hopefully, simple. Can you please explore solutions that don't have this `icon` and `description` mutable state?